### PR TITLE
Add support for AWS KMS for signing

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes GH-514

### CODE REVIEW

File: internal/signing/google_cloud_kms.go [hunk 1]
```diff
@@ -40,9 +40,5 @@ func NewGoogleCloudKMS(ctx context.Context) (KeyManager, error) {
 }
 
 func (kms *GoogleCloudKMS) NewSigner(ctx context.Context, keyID string) (crypto.Signer, error) {
-	signer, err := gcpkms.NewSigner(ctx, kms.client, keyID)
-	if err != nil {
-		return nil, err
-	}
-	return signer, nil
+	return gcpkms.NewSigner(ctx, kms.client, keyID)
```
Comment (mikehelmick) [inline internal/signing/google_cloud_kms.go:43]: wow - don't know why I did this before. duh.

Comment (sethvargo) [reply internal/signing/google_cloud_kms.go:43]: You were buffing your LOC stats 😛

**Pull request merged**